### ISSUE TITLE
[Announcement] Support for Firefox discontinued

### ISSUE THREAD

**Issue by mkiol** [mkiol/GNotifier#162]
Dear users and contributors,

I have to announce that I will not release any new version of GNotifier for Firefox. The last release is 1.9.8 (version currently available on [AMO](https://addons.mozilla.org/firefox/addon/gnotifier/)). My further development will be focused only on Thunderbird.

Here is a few reasons why support for Firefox can't be continued:
* According to [this](https://blog.mozilla.org/addons/2017/01/24/preventing-add-ons-third-party-software-from-loading-dlls-into-firefox/), starting from FF 53 (release date: April 18th), add-ons will not be allowed to use external DLLs, but GNotifier's implementation for Windows relays on DLLs. It is very likely that also js-ctypes API will be disallowed in future FF release, so integration based on libnotify on Linux will not work any more as well. 
* Mozilla is migrating Firefox add-ons framework to [WebExtensions](https://developer.mozilla.org/en-US/Add-ons/WebExtensions) and starting with FF 57 only WebExtensions will be allowed. New framework is very secure and disallows all low-level interactions between extension and the browser. Unfortunately key components of GNotifier relays on this low-level integration.
* GNotifier is less relevant these days, especially on Linux . Most Linux distributions enabled FF built-in integration with notification-server, so GNotifier is not needed any more. The only exception is Ubuntu with Unity DE, but this is temporary taking into account recent Canonical announcement that they will adopt GNOME in future Ubuntu releases.

**Comment by Lootyhoof** [comment]
Would you also consider continuing support for the Firefox version specifically for Pale Moon? Pale Moon is committed to continuing support for these types of add-ons going forward, and although support was added in #157 it appears that current development versions now do not work on it.

**Comment by mkiol** [comment]
@Lootyhoof Thank you for mentioning Pale Moon. I will try my best to not break compatibility with Pale Moon but my first priority is and will be Thunderbird. 

Please check the latest *-dev package. It should work in Pale Moon as well. My mistake. I previously used some "far too modern" javascript expression that wasn't yet supported in PM.

**Comment by Lootyhoof** [comment]
@mkiol Thank you! I can confirm this works as expected.

In order for people to be made better aware that this works in Pale Moon (as the AMO entry is [correctly] only for Thunderbird and SeaMonkey now), as well as providing a way for Pale Moon users to auto-update this add-on as required, I hope you would consider also hosting this on the [Pale Moon Add-ons Site](https://addons.palemoon.org/). If this interests you, please let me know and I will set you up and email you some details too (see [here](http://developer.palemoon.org/Add-ons:Site) for more details). Of course if you don't want to then that's fine also (maybe you have your own reasons).

Thanks again for the continued support (and sorry for clogging up your announcement post with this too)!

**Comment by mkiol** [comment]
@Lootyhoof Ooo, I didn't know that Pale Moon has now its own Add-on infrastructure! Nice :-) Yes, it interest me. Please send me additional info to gnotifier[at]mkiol.net. Thanks.

**Comment by protist** [comment]
FWIW looks like GNotifier is broken in Firefox 53.0.2 (Arch Linux).

**Comment by mkiol** [comment]
@protist The 1.9.8 version should be still working on FF 53. It is hard to download it from AMO because only the newest version is visible. [Here](https://addons.mozilla.org/firefox/downloads/file/513187/gnotifier-1.9.8-tb+fx+sm-linux.xpi?src=version-history) is direct link. I tested it on Arch with FF 53.0.2 and had no problems.

**Comment by protist** [comment]
Works great! Thank you!

**Comment by Vistaus** [comment]
Could you at least keep support for SeaMonkey? SM still doesn't use native notifications by default, it won't move to WebExtensions and there are quite a lot of Linux users (including me) who use SeaMonkey.

**Comment by mkiol** [comment]
@Vistaus Yes, SeaMonkey is on my test check list. If the API doesn't get changed too much comparing to Thunderbird, I will try to maintain compatibility with SM as well.

**Comment by Vistaus** [comment]
@mkiol Yes yes yes! That's great new! Thank you so much! 😄

**Comment by rugk** [comment]
Maybe put a note at https://addons.mozilla.org/de/thunderbird/addon/gnotifier/ that Firefox is not supported anymore (also with a small list of reasons or just "because it is built-in nowadays") and remove the badges from the Readme or so…

**Comment by rugk** [comment]
> It is hard to download it from AMO because only the newest version is visible.

No, not really. Just go to the bottom and there are the "version information" and at the bottom there is a link called "full version history". Click on it and you get to [the whole list of versions](https://addons.mozilla.org/en/thunderbird/addon/gnotifier/versions/).

**Comment by hobarrera** [comment]
Thanks for maintaining this for Firefox during all the time you did! I was unaware that now Firefox has native support (that's cool), but thanks providing and maintaining this for us [users] all that time Firefox didn't! ❤️ 💟

**Comment by robinvd** [comment]
@WhyNotHugo Where can i find the native support in firefox? i am currently not getting native notifications (on firefox 57.0a1)

**Comment by rugk** [comment]
Usually all distros/Firefox has built-in support for notifications. Open a bugzilla issue and/or an issue in your distro's bug tracker if it does not work.

**Comment by haarp** [comment]
> Where can i find the native support in firefox? i am currently not getting native notifications (on firefox 57.0a1)

I'd like to know this aswell. Currently not getting any notifications. I remember distros using modified sources for Firefox or custom extensions for integration in the past. I wonder how to make that work.

**Comment by tooomm** [comment]
>from @WhyNotHugo:
>I was unaware that now Firefox has native support 

Well, it hasn't. There is no native support for notifications on system level. On Windows at least I can't find anything like that. :(

All Firefox gives me awful specific firefox notifications:
![ff_notification](https://user-images.githubusercontent.com/9874850/32839448-c0de85c2-ca14-11e7-818d-718f1e7a12db.png)

<br>

Problem is though, that the GNotifier notifications somehow still worked all the time until recently.
A update to FF 56 (?) might broke any kind of notifications for me, and I didn't realize at first...
But now while investigating with missing notifications from another addon, I started digging.
Well, I don't download stuff all the time or use notifications from websites. So I didn't realize missing notifications at all first. :D

Funny is, that once you change the notification engine in the GNotifier settings to `Standard` (many might still have the addon installed, uninstalling might help too) the crappy mozilla notifications show up at least. Better than nothing maybe.
So check that out @haarp and @robinvd!

Looks like we are really left with only them... any other recommendations are highly appreciated! :(

**Comment by kabili207** [comment]
Mozilla is still working on adding support for the native Windows notification API, so it looks like we're stuck with the ugly Firefox notifications until then. https://bugzilla.mozilla.org/show_bug.cgi?id=1155505
